### DISCUSSION TITLE
Pubsub pipeline

### DISCUSSION THREAD

**Discussion by PedroMPagani** [xetorthio/jedis#3157] [Q&A] [answered]
I'm just wondering one thing, if I use pipelines for pubsub (even tho I'm doing 1 call each time), would I have to wait for the response of the redis server?

```java
try(Jedis jedis = getResource()){
jedis.publish(...); // waits for the complete response of the redis server?
}

try(Jedis jedis = getSoruce()){
jedis.pipelined()

// publish and sync here
}

```

Would the pipeline on pubsubs be a fire and forget?

**Comment by sazzad16** [comment, marked as answer]
> jedis.publish(...); // waits for the complete response of the redis server?

Yes.

> Would the pipeline on pubsubs be a fire and forget?

Not exactly.

---

Note: You must not use same connection for subscribe and pipeline at the same time.

**Reply by PedroMPagani** [reply]
Yeah, thanks for the note, we're already using a diff connection, at least it should because of the JedisPool? (haven't seen any crashes on the pubsub so should be working).

Is there any form of fire and forget I could use?

**Reply by sazzad16** [reply]
JedisPooled.publish(...) would give you a sense of fire and forget.

If you are going to publish a lot within a short time, you can use Pipeline.publish(...); but you would have to call Pipeline.sync() manually (after a few hundreds/thousands calls would be enough).

**Reply by PedroMPagani** [reply]
Hey, thanks.
Yeah, unfortunately I can't afford the time wait to have to write a lot of messages, the system isnt running on that much messages tho, less than 2k/s.
